### DISCUSSION TITLE
Does this crate work on mobile? (Android/iOS support)

### DISCUSSION THREAD

**Discussion by lbirkert** [kotauskas/interprocess#67] [Q&A] [answered]
Does anyone know if this crate works on mobile? Android and iOS should be UNIX based, so I wouldn't know why not. Has anyone tested it?

**Comment by kotauskas** [comment, marked as answer]
Android fulfills the `#[cfg(unix)]` gate, and conditional compilation gates for features specific to the Linux kernel are specified as `#[cfg(any(target_os = "linux", target_os = "android"))]` in Interprocess, so it's very explicitly acknowledged as a platform.

There is currently no CI for Android, so it rests firmly in the [support by association](https://github.com/kotauskas/interprocess?tab=readme-ov-file#support-by-association) tier, but [Nushell ends up testing Interprocess in its CI for Linux on ARM](https://github.com/kotauskas/interprocess/issues/63), so automatic testing is actually being done in practice in some capacity.

There is CI with tests for macOS here in this repo, but not for iOS, so it's in the same support by association tier as Android. There is currently no iOS downstream to my knowledge.

**Comment by lbirkert** [comment]
Mhmm okay. Thanks for your elaborate response!
